packages/core/src/envelope.ts

```diff
@@ -94,8 +97,11 @@ export function createEventEnvelope(
 
 /**
  * Create envelope from Span item.
+ *
+ * Takes an optional client and runs spans through `beforeSendSpan` if available.
+ * @returns A SpanEnvelope or null if all spans have been dropped
  */
-export function createSpanEnvelope(spans: SentrySpan[]): SpanEnvelope {
+export function createSpanEnvelope(spans: SentrySpan[], client?: Client): SpanEnvelope | null {
```

Comment:
_technically_ speaking this is breaking (as we export it from `@sentry/core`) 🤔 it is unlikely that a user is using this, but... hmm, not sure - maybe it is "safer" to still always return an envelope, and then check (outside) to only send this envelope if there are any spans inside of it?

Reply:
Good point, thanks!

I changed it to check the span items on the returned envelope before sending it and pulled everything down into the helper method for sending.